contents/docs/libraries/next-js.md

```diff
@@ -332,25 +335,28 @@ import { PostHog } from 'posthog-node'
 export default function PostHogClient() {
   const posthogClient = new PostHog(process.env.NEXT_PUBLIC_POSTHOG_KEY, {
     host: process.env.NEXT_PUBLIC_POSTHOG_HOST,
+    flushAt: 1,
+    flushInterval: 0
   })
   return posthogClient
 }
 ```
 
-With this component, we can both send events in the body of a component, and get data from PostHog (such as feature flag evaluations) using a server-side function (named `getData()` here).
+> **Note:** Because our server-side `posthog-node` initializations are short-lived, we set `flushAt` to `1` and `flushInterval` to `0`. `flushAt` sets how many how many capture calls we should flush the queue (in one batch). `flushInterval` sets how many milliseconds we should wait before flushing the queue. Setting them to the lowest number ensures events are sent immediately and not batched. We also need to call `await posthog.shutdownAsync()` after sending events to flush them.
```

Comment:
Much better explanation, thanks!

Im a bit confused though: If we set the flush interval and flushAt, why do we still need to call posthog.shutdownAsync() ? What's the point of setting the flushInterval and flushAt if we just shutdown manually?

Reply:
shutdownAsync also does some other clean up like stop the featureFlagsPoller which is useful.

Reply:
Reworded